SimMuon/GEMDigitizer/src/GEMPadDigiClusterProducer.cc

```diff
@@ -53,10 +53,15 @@ void GEMPadDigiClusterProducer::produce(edm::Event& e, const edm::EventSetup& ev
 }
 
 
-void GEMPadDigiClusterProducer::buildClusters(const GEMPadDigiCollection &det_pads, GEMPadDigiClusterCollection &out_clusters)
+void GEMPadDigiClusterProducer::buildClusters(const GEMPadDigiCollection &det_pads,
+                                              GEMPadDigiClusterCollection &out_clusters)
 {
+  // construct clusters
   for (const auto& ch: geometry_->chambers()) {
-    unsigned int nClusters = 0;
+
+    // proto collection
+    std::vector<std::pair<GEMDetId, GEMPadDigiCluster> > proto_clusters;
```

Comment:
hi @dildick - could you not do the cleanup of extra digis (presumably not needed all the time..) directly in out_clusters instead of making this intermediate structure?

Reply:
hi @dildick - sorry for an extremely slow answer. I'd suggest we build this idea in from the beginning.I think its straightforward

1) don't declare proto_clusters, just fill out_clusters.
2) make a dedicated function for cleaning. -in the current case, its very trivial. But it could then easily grow to meet the more complex needs in the future. (and if you need to copy clusters at that point you could).

A different point  - the way maxClusters_ is used below looks potentially incorrect. If it is really meant as the maximum size of what is now protoClusters, then don't let protoClusters grow beyond that size from the beginning. If instead its meant to be the maximum size of the final collection, then the loopmax concept is incorrect. Instead the out_clusters size needs to be checked at the end to see if it is too big.

Reply:
Regarding point 1: could you point me to an example how to clean a muondigicollection such as out_clusters?

Reply:
ah - sigh - i didn't realize that MuonDigiCollection exposes only a trivial bit of the interface that a standard vector has... i guess its not worth the trouble..

Reply:
what is the plan forward then?

Reply:
hi @dildick - could you confirm the role of maxClusters?

Reply:
max number of clusters (in the cluster digi collection) in a GEM chamber

Reply:
Ok, as a folllowup pr, the check on max clusters can be applied directly when making proto_clusters